extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergAppend.java

```diff
@@ -0,0 +1,51 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.util;
+
+import io.deephaven.annotations.BuildableStyle;
+import io.deephaven.engine.table.Table;
+import org.immutables.value.Value;
+
+import java.util.List;
+
+@Value.Immutable
+@BuildableStyle
+public abstract class IcebergAppend {
```

Comment:
I would need to make similar API changes for read operations too, like `getTableDefinition`, `getTableDefinitionTable`, and `readTable, each of which have 4 overloads.
Is that okay?
Also, any other APIs which you would want me to change like `listTablesAsTable`, `listSnapshotsAsTable`, `listNamespacesAsTable`, `listNamespaces`, which have 2 or 3 overloads but much simpler.

cc: @devinrsmith